sycl/plugins/unified_runtime/ur/adapters/level_zero/common.hpp

```diff
@@ -309,14 +298,22 @@ ur_result_t ze2urResult(ze_result_t ZeResult);
 // Trace a call to Level-Zero RT
 #define ZE2UR_CALL(ZeName, ZeArgs)                                             \
   {                                                                            \
-    ze_result_t ZeResult = ZeName ZeArgs;                                      \
-    if (auto Result = ZeCall().doCall(ZeResult, #ZeName, #ZeArgs, true))       \
+    urPrint("ZE ---> %s%s\n", #ZeName, #ZeArgs);                               \
+    if (auto Result = ZeCall().doCall(ZeName ZeArgs, #ZeName, #ZeArgs))        \
       return ze2urResult(Result);                                              \
   }
 
-// Perform traced call to L0 without checking for errors
+#define ZE_CALL(ZeName, ZeArgs, ZeResult)                                      \
+  {                                                                            \
+    urPrint("ZE ---> %s%s\n", #ZeName, #ZeArgs);                               \
+    ZeResult = ZeCall().doCall(ZeName ZeArgs, #ZeName, #ZeArgs);               \
+  }
+
 #define ZE_CALL_NOCHECK(ZeName, ZeArgs)                                        \
-  ZeCall().doCall(ZeName ZeArgs, #ZeName, #ZeArgs, false)
+  {                                                                            \
+    urPrint("ZE ---> %s%s\n", #ZeName, #ZeArgs);                               \
+    ZeCall().doCall(ZeName ZeArgs, #ZeName, #ZeArgs);                          \
+  }
```

Comment:
why keep the ZE_CALL_NOCHECK?

Reply:
ZE_CALL_NOCHECK is being used on methods that don't return a return value, so that's the different between ZE_CALL and ZE_CALL_NOCHECK. Both dont check for result (as before), but ZE_CALL_NOCHECK doesn't return the return value.

One idea here would be to merge both, and add a [maybe_unused] to result, for when this macro is used on methods that don't have a return value. what do you think?

Reply:
I think this is a reasonable idea.

Reply:
@smaslov-intel : to avoid having too many rewrites, I have created at the end 3 macros, as follows:

```cpp
// Trace a call to Level-Zero RT and
// return if error has occurred.
#define ZE2UR_CALL(ZeName, ZeArgs)                                             \
  {                                                                            \
    urPrint("ZE ---> %s%s\n", #ZeName, #ZeArgs);                               \
    if (auto Result = ZeCall().doCall(ZeName ZeArgs, #ZeName, #ZeArgs))        \
      return ze2urResult(Result);                                              \
  }

// Trace a call to Level-Zero RT without
// checking for call's result and instead
// return it in ZeResult declared
// in original scope
#define ZE_CALL_NOCHECK(ZeName, ZeArgs, ZeResult)                              \
  {                                                                            \
    urPrint("ZE ---> %s%s\n", #ZeName, #ZeArgs);                               \
    ZeResult = ZeCall().doCall(ZeName ZeArgs, #ZeName, #ZeArgs);               \
  }

// Trace a call to Level-Zero RT and
// return without checking for call's
// result
#define ZE_CALL_NOCHECK_VOID(ZeName, ZeArgs)                                   \
  {                                                                            \
    urPrint("ZE ---> %s%s\n", #ZeName, #ZeArgs);                               \
    ZeCall().doCall(ZeName ZeArgs, #ZeName, #ZeArgs);                          \
  }

```